### DISCUSSION TITLE
MP3 encoding is not working properly

### DISCUSSION THREAD

**Discussion by Alexejvr** [enzo1982/freac#580] [Audio Format Discussion]
I used Fre:ac since long time but only recently I compare the mp3 compression results with Spek program and it seems not working fine.
I started from classical flac file. Then I generate two mp3 CBR 320 kbps with maximum quality but one with Fre:ac and one with winLAME (both with same version of LAME). The spectrum of the two mp3 is very similar also if different from FLAC ones. And this is regular.
Then I have done the same test with another FLAC file with rock music. The result is not the same before: FLAC is better but the two mp3 are slightly different (see attached pictures, Better is winLame and worst is Fre:ac). If I do it again with a pop FLAC the difference is bigger but always winLAME is better than Fre:ac. Why ?  
Classical
FLAC
![image](https://github.com/enzo1982/freac/assets/155686471/ada28c58-c98c-4b8d-816c-727de0b49eec)
Fre:ac
![image](https://github.com/enzo1982/freac/assets/155686471/bc51012d-e459-4b0d-a4bd-927f5c5a0041)
winLAME
![image](https://github.com/enzo1982/freac/assets/155686471/2eff4a05-79ab-4587-952d-280af0c85612)

Rock
FLAC
![image](https://github.com/enzo1982/freac/assets/155686471/6ec413f6-ff47-4faa-aeef-82c0f9665def)
MP3 Fre:ac
![image](https://github.com/enzo1982/freac/assets/155686471/5f36e309-bc18-44f6-8d92-376d519d3b5c)
MP3 winLAME
![image](https://github.com/enzo1982/freac/assets/155686471/01bb9746-4bad-46c9-a99e-eac3d5ee5a32)

Pop
FLAC
![image](https://github.com/enzo1982/freac/assets/155686471/fbad0c25-d872-48a4-95de-227650a6b8d6)
Fre:ac
![image](https://github.com/enzo1982/freac/assets/155686471/094ec2b1-5ad8-4fa5-89a3-8f79f579e08c)
winLAME
![image](https://github.com/enzo1982/freac/assets/155686471/cfc90868-d4ff-4cb3-b271-ff694ffaec63)

**Comment by enzo1982** [comment]
Your Spek screenshots do not indicate any issue with fre:ac. Quite the contrary, it rather looks like there may be an issue with your winLAME settings.

This is due to how the MP3 format works. The format is limited to a maximum bitrate of 320 kbps which is not enough to accurately represent all frequencies present in most rock and pop songs. Therefore, MP3 codecs try to identify frequencies and sounds which will not be audible to the human ear. These frequencies are then filtered out which leaves more of the available 320 kbps available to more accurately represent the remaining actually audible frequencies.

Your winLAME screenshots look like it is wasting a lot of bitrate on sounds between 16 and 20 kHz which will be inaudible in most cases. fre:ac, however, filters most of the sounds in that frequency range and will use the freed up bits to more accurately represent sounds in the lower frequency ranges. Note though that it still lets some sounds in the higher frequency range pass through. In the pop file screenshots you can even see that it let's some frequencies above 20 kHz pass through which are filtered out by winLAME (which seems to be configured to use a hard cut-off at 20 kHz). That's because when configured correctly, LAME does not use any hard cut-offs but analyzes the sound with so called psycho-acoustic algorithms to identify sounds that may have a chance to be audible even at such extremely high frequencies.

Your winLAME screenshots look like psycho-acoustics might be disabled or misconfigured.

By the way, you can definitely configure fre:ac to act similarly - e.g. by checking the _Disable all filtering_ option in the LAME audio processing settings. fre:ac will then produce MP3s that look really nice in Spek but sound much worse when you actually listen to them. But if you listen with your ears instead of your eyes, I'd recommend to leave filtering enabled.

**Reply by Alexejvr** [reply]
Thank you for your reply.
I agree with most of your writing.
I kindly ask you to look again to spectrum images and their labels.
If you look better to images the Fre:ac spectrum seems worst than winLAME: I attach again two images, first image is Fre:ac and second winLAME 
![image](https://github.com/enzo1982/freac/assets/155686471/7a3dc943-3a40-434d-8f28-bf98ac67379d)
![image](https://github.com/enzo1982/freac/assets/155686471/c1c3e25e-d264-4814-a001-98d4c13634cf)

About settings: here there are the ones used for Fre:ac
![image](https://github.com/enzo1982/freac/assets/155686471/8c0af353-e561-4629-b9f6-12ebf8b179be)
![image](https://github.com/enzo1982/freac/assets/155686471/30999f6d-e703-425a-8fa0-7cf611bdb68c)
![image](https://github.com/enzo1982/freac/assets/155686471/bf47553b-1a17-4599-a8df-f95f004f5397)
![image](https://github.com/enzo1982/freac/assets/155686471/78d078a6-3de9-4cc7-8328-065f472e7785)
I tried to avoid any filter influence or psycho-acoustic.
WinLAME use the "-b 320 --cbr" LAME library parameters.

I also tried to uninstall and install again the fre:ac, to use a portable version of Fre:ac but result is the same.

So I decided to make again the encoding with Fre:ac changing one parameter by one and the final result is that problem is related to usage of Stereo/Joint Stereo parameter.
Here there are the two spectrums and the media info:
Stereo (with issue)
![image](https://github.com/enzo1982/freac/assets/155686471/07231df5-d082-4967-844c-634145804ead)
Audio
Format                                   : MPEG Audio
Format version                           : Version 1
Format profile                           : Layer 3
Duration                                 : 4min 15s
Bit rate mode                            : Costante
Bit rate                                 : 320 kb/s
Channel(s)                               : 2 canali
Sampling rate                            : 44,1 KHz
Frame rate                               : 38,281 fps (1152 SPF)
Compression mode                         : Con perdita
Replay gain                              : -6,75 dB
Replay gain peak                         : 1.025639
Stream size                              : 9,76 MiB (99%)
Writing library                          : LAME3.100
Encoding settings                        : -m s -V 4 -q 3

Joint Stereo (no problem)
![image](https://github.com/enzo1982/freac/assets/155686471/62194c5d-2529-4355-a541-e5ec62c8bdd8)
Audio
Format                                   : MPEG Audio
Format version                           : Version 1
Format profile                           : Layer 3
Format settings                          : Joint stereo / MS Stereo
Duration                                 : 4min 15s
Bit rate mode                            : Costante
Bit rate                                 : 320 kb/s
Channel(s)                               : 2 canali
Sampling rate                            : 44,1 KHz
Frame rate                               : 38,281 fps (1152 SPF)
Compression mode                         : Con perdita
Replay gain                              : -6,75 dB
Replay gain peak                         : 1.025639
Stream size                              : 9,76 MiB (99%)
Writing library                          : LAME3.100
Encoding settings                        : -m j -V 4 -q 3

What do you think about it ?
